Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -90,6 +90,22 @@ extension File {
         }
     }
 
+    public func syntaxKindsByLine(startLine: Int? = nil,
+                                  endLine: Int? = nil) -> [(Int, [SyntaxKind])] {
+        let contents = self.contents as NSString
+        let kindsWithLines = syntaxMap.tokens.map { token -> (Int, SyntaxKind) in
+            let tokenLine = contents.lineAndCharacterForByteOffset(token.offset)
+            return (tokenLine!.line, SyntaxKind(rawValue: token.type)!)
```

Comment:
This is because currently `.filter` needs the line. If we filtered first, `lineAndCharacterForByteOffset` would be called twice.